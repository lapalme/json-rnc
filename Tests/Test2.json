{
  "address": {
    "streetAddress": "21 2nd Street",
    "city": "New York"
  },
  "phoneNumber": [
    {
      "location": "home",
      "code": 44
    }
  ],
  "code":"allo"
}

{"address":{"city":"Montréal","streetAddress":"St-Laurent"},
 "phoneNumber":[{"location":"office","code":47493}],
 "code":"bonjour"
}

{"address":{"city":"Québec"},
 "phoneNumber":[{"location":"home","code":345},{"location":"mobile","code":false}],
 "code":"allo"
}
 
{}
[345]
"bonjour"
{"address":{"city":"Paris"},"phoneNumber":[]}
