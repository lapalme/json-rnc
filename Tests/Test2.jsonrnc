##  schéma correspondant à l'exemple http://jsonschema.net/#/home
debut = {
    address:{ streetAddress?: string, city: string},
    phoneNumber:[{location:string, code:integer}]
}
start=debut
