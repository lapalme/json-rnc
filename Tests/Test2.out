3:{u'code': u'allo', u'phoneNumber': [{u'code': 345, u'location': u'home'}, {u'code': Fal...\xe9bec'}}
phoneNumber/[1]/code	integer expected:	false
4:{}
	missing required field:address	
	missing required field:phoneNumber	
5:[345]
	object expected:	[345]
6:bonjour
	object expected:	bonjour
7 objects read: 4 invalid, 0 bad, 0 with duplicate fields
Error Statistics
              2	:object expected:
              1	:missing required field:address
              1	phoneNumber/[1]/code:integer expected:
              1	:missing required field:phoneNumber
